test/test.hs

```diff
@@ -106,6 +107,134 @@ macroTests = testGroup "macroTests" $
+                                 (stringToKeyMacro "'c'v")))
+         @?= "xyyzxyyzxyyzxyyzxyyxyyzxyyzxyyzxyyzxyy"
+     , testCase "RepeatLast test 29" $
+         snd (last (unwindMacros (bindInput [ ("a", "'xy'v")
+                                            , ("b", "'za'V")
+                                            , ("c", "'ab'V") ] coinput)
+                                 (stringToKeyMacro "'c'v")))
+         @?= "xyyzxyyzxyyxyyzxyyzxyyxyyzxyyzxyyxyyzxyyzxyy"
+     , testCase "RepeatLast test 30" $
+         snd (last (unwindMacros (bindInput [ ("a", "'xy'v")
+                                            , ("b", "'za'V")
+                                            , ("c", "'ab'V") ] coinput)
+                                 (stringToKeyMacro "'c'V")))
+         @?= "xyyzxyyzxyyxyyzxyyzxyyxyyzxyyzxyyxyyzxyyzxyy"
+     -- , testCase "RepeatLast test 31" $
+     --     snd (last (unwindMacros (bindInput [ ("a", "'xy'v")
+     --                                        , ("b", "'za'v")
+     --                                        , ("c", "'ab'V") ] coinput)
+     --                             (stringToKeyMacro "'c'V")))
+     --     @?= "xyyzxyyxyyzxyyxyyzxyyxyyzxyy"
+     -- , testCase "RepeatLast test 32" $
+     --     snd (last (unwindMacros (bindInput [ ("a", "'xy'v")
+     --                                        , ("b", "'za'v") ] coinput)
+     --                             (stringToKeyMacro "'ab'vv")))
+     --     @?= "xyyzxyyxyyzxyyxyyxyyzxyyxyyzxyyxyy"
+     -- , testCase "RepeatLast test 33" $
+     --     snd (last (unwindMacros (bindInput [ ("a", "'xy'V") ] coinput)
+     --                             (stringToKeyMacro "a'za'vvv")))
+     --     @?= "xyxyzxyxyxyxyzxyxyxyxyxyxyzxyxyxyxyzxyxyxyxy"
+     -- , testCase "RepeatLast test 34" $
+     --     snd (last (unwindMacros (bindInput [ ("a", "'xy'v")
+     --                                        , ("c", "a'za'Vv") ] coinput)
+     --                             (stringToKeyMacro "'c'v")))
+     --     @?= "xyyzxyyzxyyzxyyzxyyxyyzxyyzxyyzxyyzxyy"
+     -- , testCase "RepeatLast test 35" $
+     --     snd (last (unwindMacros (bindInput [ ("a", "'xy'v")
+     --                                        , ("b", "'za'V") ] coinput)
+     --                             (stringToKeyMacro "'ab'Vv")))
+     --     @?= "xyyzxyyzxyyxyyzxyyzxyyxyyzxyyzxyyxyyzxyyzxyy"
+     , testCase "RepeatLast test 36" $
+         snd (last (unwindMacros (bindInput [ ("a", "'xy'v")
+                                            , ("b", "za'za'") ] coinput)
+                                 (stringToKeyMacro "'ab'V'ab'V")))
+         @?= "xyyzxyyzxyyxyyzxyyzxyyxyyzxyyzxyyxyyzxyyzxyy"
+     -- , testCase "RepeatLast test 37" $
+     --     snd (last (unwindMacros (bindInput [ ("b", "z'xy'vv")
+     --                                        , ("c", "'xyvb'V") ] coinput)
+     --                             (stringToKeyMacro "'c'V")))
+     --     @?= "xyyzxyyxyyzxyyxyyzxyyxyyzxyy"
```

Comment:
Whoah, this case gave me a headache. Turns out that `v` is being saved when recording a macro (we don't filter this in `promptGetKey`!). This means when we repeat the last action when recording in-game macro, it gets repeated twice when repeating the macro.

Reply:
Does it mean my cat was right? Anyway, I would understand better if we had a minimal test that exhibits exactly the behaviour you describe.

Reply:
It was onto something. Fixing this scenario caused 17 other cases to fail, so this made me thinking.